examples/wandb_sweeps/sweep.py

```diff
@@ -29,22 +29,48 @@ def train_sweep(
         argv: command-line arguments.
     """
     populate_config(config, temp_config)
-    run_sweep(argv)
+    exit_code = run_sweep(argv)
+    wandb.finish(exit_code=exit_code)
 
 
-def run_sweep(argv: List[str]) -> None:
+def run_sweep(argv: List[str]) -> int:
     """Actually runs the sweep.
 
     Args:
         argv: command-line arguments.
 
+    Returns:
+        int: The exit code from the subprocess. 0 for success.
+
     We encapsulate each run by using a separate subprocess, which ought to
     ensure that memory is returned (etc.).
     """
     try:
-        subprocess.check_call(argv)
-    except subprocess.CalledProcessError as error:
-        logging.error("Subprocess error: %s", error)
+        process = subprocess.Popen(argv, stderr=subprocess.PIPE)
+
+        while True:
+            # Read stderr
+            line = process.stderr.readline()
+            if line:
+                # Decode bytes and strip any trailing whitespace
+                error_msg = line.decode("utf-8").rstrip()
+                if error_msg:  # Only log non-empty messages
+                    logging.info(error_msg)
+
+            # Check if process has finished
+            if line == b"" and process.poll() is not None:
+                break
+
+        if process.returncode != 0:
+            error_msg = f"Subprocess' exit status {process.returncode}"
+            logging.error(error_msg)
+
+        return process.returncode
+
+    except Exception:
+        error_msg = f"Full traceback: {traceback.format_exc()}"
+        logging.error(error_msg)
```

Comment:
I think it's simpler to just one line this (it's self documenting that it's an error message)

logging.error(f"Full traceback: {traceback.format_exc()}")

Same comment for 65,66